sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_discovery.py

```diff
@@ -0,0 +1,104 @@
+# ------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -------------------------------------------------------------------------
+
+import os
+import time
+from dataclasses import dataclass
+import dns.resolver
+from dns.resolver import NXDOMAIN, LifetimeTimeout, NoNameservers  # cspell:disable-line
+from dns.rdatatype import SRV  # cspell:disable-line
+from ._constants import DISABLE_APPCONFIGURATION_DISCOVERY, HTTPS_PREFIX
+
+
+@dataclass
+class SRVRecord:
+    priority: int
+    weight: int
+    port: int
+    target: str
+    protocol: str = HTTPS_PREFIX
+
+    def __init__(self, answer):
+        self.priority = answer.priority
+        self.weight = answer.weight
+        self.port = answer.port
+        self.target = str(answer.target).rstrip(".")
+
+
+def find_auto_failover_endpoints(endpoint):
```

Comment:
Missing type annotations - intentional?

Reply:
Not really, I'm assuming this wasn't flagged as it's not a public API.